
syntax = "proto3";

package helloworld;

service Greeter {
  
  rpc SayHello (HelloRequest) returns (HelloReply) {}

  rpc Signup (LoginRequest) returns (LoginResponse) {}

  rpc Login (LoginRequest) returns (LoginResponse) {}

  rpc Logout (TokenInfo) returns (LoginResponse) {}

}

message LoginRequest {
  string name = 1;
  string password = 2;
  string device = 3;
}

message LoginResponse {
  int32 code = 1;
  string message = 2;
  TokenInfo tokenInfo = 3;
}

message TokenInfo {
  string token = 1;
}

message HelloRequest {
  string name = 1;
}

message HelloReply {
  string message = 1;
}
